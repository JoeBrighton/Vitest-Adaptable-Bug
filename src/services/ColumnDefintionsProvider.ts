import { ColDef, ColGroupDef } from '@ag-grid-enterprise/all-modules';

import ColumnConfiguration from './ColumnConfiguration.json';

export default class ColumnDefinitionsProvider {
    private readonly columnDefinitions: (ColDef | ColGroupDef)[] | null | undefined;

    constructor() {
        this.columnDefinitions = this.loadColumnDefinitions();
    }

    public getColumnDefinitions(): (ColDef | ColGroupDef)[] | null | undefined {
        return this.columnDefinitions;
    }

    private loadColumnDefinitions(): (ColDef | ColGroupDef)[] | null | undefined {
        const columnDefinitions: ColDef[] = ColumnConfiguration;

        const updatedCols: ColDef[] = [];
        columnDefinitions.forEach((colDef) => {
            updatedCols.push(this.addDefaultProperties(colDef));
        });

        return updatedCols;
    }

    public addDefaultProperties(columnDefintion: ColDef): ColDef {
        const result = {
            ...columnDefintion,
            filterParams: {
                cellHeight: 25
            }
        };
        return result;
    }
}

export const columnDefinitionsProvider = new ColumnDefinitionsProvider();