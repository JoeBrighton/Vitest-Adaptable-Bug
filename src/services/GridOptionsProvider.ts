import { AdaptableToolPanelAgGridComponent } from '@adaptabletools/adaptable/src/AdaptableComponents';
import { GridOptions } from '@ag-grid-enterprise/all-modules';
import { columnDefinitionsProvider } from './ColumnDefintionsProvider';

export default class GridOptionsProvider {
    private readonly gridOptions: GridOptions;

    constructor() {
        this.gridOptions = {
            columnDefs: columnDefinitionsProvider.getColumnDefinitions(),
            sideBar: true,
            components: {
                AdaptableToolPanel: AdaptableToolPanelAgGridComponent
            },
            suppressMenuHide: true,
            enableRangeSelection: true
        };
    }

    public getGridOptions(): GridOptions {
        return this.gridOptions;
    }
}

export const gridOptionsProvider = new GridOptionsProvider();