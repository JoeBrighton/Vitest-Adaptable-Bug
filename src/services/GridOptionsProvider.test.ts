import GridOptionsProvider from './GridOptionsProvider';
import { columnDefinitionsProvider } from './ColumnDefintionsProvider';

describe('GridOptionsProvider', () => {
    it('should provide grid options', () => {
        const columnDefinitionsProviderSpy = jest.spyOn(columnDefinitionsProvider, 'getColumnDefinitions');
        columnDefinitionsProviderSpy.mockReturnValue([]);
        const sut = new GridOptionsProvider();
        const result = sut.getGridOptions();
        expect(result).toMatchSnapshot();
    });
});
