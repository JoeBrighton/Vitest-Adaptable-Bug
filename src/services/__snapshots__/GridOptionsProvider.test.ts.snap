// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridOptionsProvider should provide grid options 1`] = `
Object {
  "columnDefs": Array [],
  "components": Object {
    "AdaptableToolPanel": [Function],
  },
  "enableRangeSelection": true,
  "sideBar": true,
  "suppressMenuHide": true,
}
`;
